### PR TITLE
test(cache): Improve tests for CacheStore

### PR DESCRIPTION
**Pull request by SEOKKAMONI**
# Overview
-  add a test for the reset method
- separate the methods using describe.

## PR Checklist

- [x] I did below actions if need

1. I read the [Contributing Guide](https://github.com/toss/suspensive/blob/main/CONTRIBUTING.md)
2. I added documents and tests.

### CODE REVIEW

File: packages/cache/src/Cache.spec.tsx [hunk 1]
```diff
@@ -2,10 +2,15 @@ import { Suspense } from '@suspensive/react'
 import { TEXT } from '@suspensive/utils'
 import { render, screen } from '@testing-library/react'
 import { Cache } from './Cache'
+import { cacheOptions } from './cacheOptions'
 import { CacheStore } from './CacheStore'
 import { CacheStoreProvider } from './CacheStoreProvider'
 
-const key = (id: number) => ['key', id] as const
+const cache = (id: number) =>
```
Comment (manudeli) [inline packages/cache/src/Cache.spec.tsx:9]:
```suggestion
const successCache = (id: number) =>
```

File: packages/cache/src/CacheStore.spec.tsx [hunk 2]
```diff
@@ -32,98 +32,121 @@ describe('CacheStore', () => {
...
-    }
-    try {
-      cacheStore.suspend(errorCache(2))
-    } catch (promiseToSuspense) {
-      await promiseToSuspense
-    }
-    expect(cacheStore.getError(errorCache(1))).toBe(ERROR_MESSAGE)
-    expect(cacheStore.getError(errorCache(2))).toBe(ERROR_MESSAGE)
+  describe('clearError', () => {
+    it("have clearError method without key should clear promise & error for all key's cached", async () => {
```
Comment (manudeli) [inline packages/cache/src/CacheStore.spec.tsx:36]:
```suggestion
    it("should clear promise & error for all cached", async () => {
```

File: packages/cache/src/CacheStore.spec.tsx [hunk 3]
```diff
@@ -32,98 +32,121 @@ describe('CacheStore', () => {
...
-    cacheStore.clearError()
-    expect(cacheStore.getError(errorCache(1))).toBeUndefined()
-    expect(cacheStore.getError(errorCache(2))).toBeUndefined()
-  })
+      cacheStore.clearError()
+      expect(cacheStore.getError(errorCache(1))).toBeUndefined()
+      expect(cacheStore.getError(errorCache(2))).toBeUndefined()
+    })
+
+    it("have clearError method with key should clear promise & error for key's cached", async () => {
```
Comment (manudeli) [inline packages/cache/src/CacheStore.spec.tsx:57]:
```suggestion
    it("should clear promise & error for specific cached", async () => {
```

File: packages/cache/src/CacheStore.spec.tsx [hunk 4]
```diff
@@ -32,98 +32,121 @@ describe('CacheStore', () => {
...
+  })
 
-    cacheStore.clearError(errorCache(1))
-    expect(cacheStore.getError(errorCache(1))).toBeUndefined()
-    expect(cacheStore.getError(errorCache(2))).toBe(ERROR_MESSAGE)
-    cacheStore.clearError(errorCache(2))
-    expect(cacheStore.getError(errorCache(1))).toBeUndefined()
-    expect(cacheStore.getError(errorCache(2))).toBeUndefined()
+  describe('remove', () => {
+    it('should take a key and remove the cached data for that key', async () => {
```
Comment (manudeli) [inline packages/cache/src/CacheStore.spec.tsx:93]:
```suggestion
    it('should remove specific cached', async () => {
```

File: packages/cache/src/CacheStore.spec.tsx [hunk 5]
```diff
@@ -32,98 +32,121 @@ describe('CacheStore', () => {
...
+      try {
+        cacheStore.suspend(successCache(1))
+      } catch (promiseToSuspense) {
+        await promiseToSuspense
+      }
+      let cached = cacheStore.getData(successCache(1))
+      expect(cached).toBe(TEXT)
+      cacheStore.reset(successCache(1))
+      cached = cacheStore.getData(successCache(1))
+      expect(cached).toBeUndefined()
```
Comment (manudeli) [inline packages/cache/src/CacheStore.spec.tsx:118]: We should differ cached.data and cached

```suggestion
      let cachedData = cacheStore.getData(successCache(1))
      expect(cachedData).toBe(TEXT)
      cacheStore.reset(successCache(1))
      cachedData = cacheStore.getData(successCache(1))
      expect(cachedData).toBeUndefined()
```

File: packages/cache/src/CacheStore.spec.tsx [hunk 6]
```diff
@@ -32,98 +32,121 @@ describe('CacheStore', () => {
...
-      <CacheStoreProvider store={cacheStore}>
-        <Suspense fallback={FALLBACK}>
-          <Cache {...errorCache(1)} cacheFn={() => sleep(ms('0.1s')).then(() => TEXT)}>
-            {(cached) => <>{cached.data}</>}
-          </Cache>
-        </Suspense>
-      </CacheStoreProvider>
-    )
+  describe('getData', () => {
+    it("have getData method with key should get data of key's cached", async () => {
```
Comment (manudeli) [inline packages/cache/src/CacheStore.spec.tsx:123]:
```suggestion
    it("should get data of specific cached", async () => {
```

File: packages/cache/src/CacheStore.spec.tsx [hunk 7]
```diff
@@ -32,98 +32,137 @@ describe('CacheStore', () => {
...
+        cacheStore.suspend(successCache(2))
+      } catch (promiseToSuspense) {
+        await promiseToSuspense
+      }
+      expect(cacheStore.getData(successCache(1))).toBe(TEXT)
+      expect(cacheStore.getData(successCache(2))).toBe(TEXT)
+      cacheStore.reset()
+      expect(cacheStore.getData(successCache(1))).toBeUndefined()
+      expect(cacheStore.getData(successCache(2))).toBeUndefined()
+    })
```
Comment (SEOKKAMONI) [inline packages/cache/src/CacheStore.spec.tsx:124]: I've also added a test case to reset all cached data.

File: packages/cache/src/CacheStore.spec.tsx [hunk 8]
```diff
@@ -32,98 +32,137 @@ describe('CacheStore', () => {
...
-    } catch (promiseToSuspense) {
-      await promiseToSuspense
-    }
-    expect(cacheStore.getData(successCache(1))).toBe(TEXT)
-    expect(() => {
-      cacheStore.remove(successCache(1))
-    }).not.throw()
-    expect(cacheStore.getData(successCache(1))).toBeUndefined()
+  describe('reset', () => {
+    it('should reset data of all cached', async () => {
```
Comment (manudeli) [inline packages/cache/src/CacheStore.spec.tsx:108]:
```suggestion
    it('should delete all cached and notify to subscribers', async () => {
```

Add test code to check "reset" do notifying to subscribers after deleting all cached

File: packages/cache/src/CacheStore.spec.tsx [hunk 9]
```diff
@@ -32,98 +32,137 @@ describe('CacheStore', () => {
...
+        await promiseToSuspense
+      }
+      expect(cacheStore.getData(successCache(1))).toBe(TEXT)
+      expect(cacheStore.getData(successCache(2))).toBe(TEXT)
+      cacheStore.reset()
+      expect(cacheStore.getData(successCache(1))).toBeUndefined()
+      expect(cacheStore.getData(successCache(2))).toBeUndefined()
+    })
+
+    it('should reset data of specific cached', async () => {
```
Comment (manudeli) [inline packages/cache/src/CacheStore.spec.tsx:126]: It's same

File: packages/cache/src/CacheStore.spec.tsx [hunk 10]
```diff
@@ -32,98 +32,137 @@ describe('CacheStore', () => {
...
+      expect(screen.queryByText(FALLBACK)).not.toBeInTheDocument()
+      expect(cacheStore.getData(errorCache(1))).toBe(TEXT)
+    })
   })
 
-  it('should handle unsubscribe gracefully when no subscribers exist', () => {
-    const mockSync = vi.fn()
-    const cacheKey = ['nonexistent', 'key'] as const
-    cacheStore.unsubscribe({ cacheKey }, mockSync)
+  describe('unsubscribe', () => {
```
Comment (manudeli) [inline packages/cache/src/CacheStore.spec.tsx:159]: How about below 2 things after this Pull Request
1. Remove unsubscribe method
2. CacheStore extends Subscribable https://github.com/TanStack/query/blob/main/packages/query-core/src/subscribable.ts

**Review by manudeli — APPROVED** [review]
@SEOKKAMONI Could you do https://github.com/toss/suspensive/pull/1147/files#r1693266363 please

**Pull request merged**